controllers/maascluster_controller.go

```diff
@@ -276,16 +278,20 @@ func (r *MaasClusterReconciler) reconcileNormal(_ context.Context, clusterScope
 
 	dnsService := dns.NewService(clusterScope)
 
-	if err := dnsService.ReconcileDNS(); err != nil {
-		clusterScope.Error(err, "failed to reconcile load balancer")
-		conditions.MarkFalse(maasCluster, infrav1beta1.DNSReadyCondition, infrav1beta1.DNSFailedReason, clusterv1.ConditionSeverityError, err.Error())
-		return reconcile.Result{}, err
-	}
+	// Only reconcile DNS resource creation if DNS name is not set or DNS is not ready
+	dnsReady := conditions.IsTrue(maasCluster, infrav1beta1.DNSReadyCondition)
+	if maasCluster.Status.Network.DNSName == "" || !dnsReady {
```

Comment:
With this condition, future DNS failures will not be reflected in the status correctly. How do we plan to handle such cases?